### ISSUE TITLE
Cannot find module for a peerDependency of a devDependency

### ISSUE THREAD

**Issue by aecz** [rstacruz/pnpm#394]
### Code to reproduce the issue:
```
npm i -g angular-cli
ng new myproject --skip-npm
cd myproject
pnpm i
ng build
```
EDIT:
Faster way to reproduce, thanks to @zkochan :
```
git clone https://github.com/zkochan/pnpm-issue394.git
cd pnpm-issue394
ng build
```

### Expected behavior:
A folder dist is created in the project directory containing the app to deploy.

### Actual behavior:
The command fails with the error: **Cannot find module 'typescript'**

### Additional information:
```
Error: Cannot find module 'typescript'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\aecz\.pnpm-store\@ngtools+webpack@1.0.0\_\src\plugin.js:2:10)
```
Typescript dependency is correctly linked in myproject/node_modules but @ngtools/webpack do not "see" it and declares it as a peerDependency. Maybe because it is itself a dependency of angular-cli. Note that angular-cli is also correctly linked locally as the project declares it as devDependency.

Node v6.5.0 on Windows 7

**Comment by zkochan** [comment]
I have some different errors but there's an issue for sure.

Code to repro https://github.com/zkochan/pnpm-issue394

**Comment by aecz** [comment]
If you dont have a global store, then another error appear indeed:
```
Error: Cannot find module '@angular/tsc-wrapped/src/symbols'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\aecz\pnpm-issue394\node_modules\.store\@angular-cli+ast-tools@1.0.2\_\src\ast-utils.js:4:17)
```
Although it works fine using npm install :(
It seems that a sub dependency does not see its siblings. With npm3 and the flat tree this issue is solved.

**Comment by aecz** [comment]
I think i found some clues. Scoped dependencies in `.store/node_modules` are as follow `@angular%2ftsc-wrapped`. If we created the folder ´@angular´ and put the links in it, solved ;) Ive never looked at the code but i can help if you points me towards the correct direction.

**Comment by zkochan** [comment]
It would be awesome you'd be able to fix this.

The code is [here](https://github.com/rstacruz/pnpm/blob/master/src/install/linkPeers.ts). Line 49, I guess

**Comment by zkochan** [comment]
I have tried your suggested fix, @aecz (branch issue394). Now the error is different - https://gist.github.com/zkochan/82201d5f69918b9525ed54f9aa0af174

**Comment by aecz** [comment]
Yes, i had this issue too. We need to also check the src of relSymlink. When it comes relative `../` for a scoped package we should set it to `../../`. It should works after that. Ive got a working version a pnpm locally. I only modified forceSymlink. Didnt know if escapedName could be removed.

**Comment by zkochan** [comment]
wow, you are right! good catch!

**Comment by zkochan** [comment]
My yesterday's error was because of this bug - https://github.com/rstacruz/pnpm/pull/399. After building native addons were fixed, there's a new error: 

```
zkochan@zkochan:~/src/test/issue394$ ng build
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
4849ms building modules                                                                  
28ms sealing
0ms optimizing 
0ms basic module optimization 
77ms module optimization
0ms advanced module optimization 
10ms basic chunk optimization       
0ms chunk optimization 
4ms advanced chunk optimization 
0ms module and chunk tree optimization 
109ms module reviving
4ms module order optimization 
2ms module id optimization 
2ms chunk reviving 
1ms chunk order optimization 
10ms chunk id optimization
61ms hashing
0ms module assets processing 
131ms chunk assets processing
2ms additional chunk assets processing 
0ms recording 
0ms additional asset processing 
1660ms chunk asset optimization
 94% asset optimizationError in bail mode: [default] /home/zkochan/src/test/issue394/src/app/app.component.ts:5:12 
Cannot find name 'require'.
```

**Comment by aecz** [comment]
Thanks for the PR. The error you are having is because typescript is looking at node_modules/@typings folder and does not find the node folder. With npm 3, the tree is flat so node typings are found at the top level mode_modules. I guess if in the package.json the engine is set to npm>=3 we should have a flat tree. Would that require a lot of effort ?

**Comment by zkochan** [comment]
Please create a separate issue for this idea of yours.

But I am afraid this won't work even with a flat structure. Currently TypeScript can't work symlinked type declaration files. That's why I couldn't use them in pnpm... Hopefully they will fix it in TypeScript 2.1
